gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/SingleLinkedListConfigKeyPath.java

```
@@ -0,0 +1,57 @@
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigKeyPath;
+
+public class SingleLinkedListConfigKeyPath implements ConfigKeyPath{
+  
+  public static final String PATH_DELIMETER = "/";
+  public static final SingleLinkedListConfigKeyPath ROOT = new SingleLinkedListConfigKeyPath(null, "");
+
+  private final ConfigKeyPath parent;
+  private final String ownName;
+  
+  // constructor private, can only create path from ROOT using createChild method
+  private SingleLinkedListConfigKeyPath(ConfigKeyPath parent, String name){
+    this.parent = parent;
+    this.ownName = name;
+  }
+  
+  @Override
+  public ConfigKeyPath getParent() {
+    return this.parent;
```

This should throw an InvalidOperationException for ROOT.